doc/website/blog/tmp-versioned-migrations-sum-file.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Versioned Migrations Management and Migration Directory Integrity
+author: MasseElch
+authorURL: "https://github.com/masseelch"
+authorImageURL: "https://avatars.githubusercontent.com/u/12862103?v=4"
+image: "TBD"
+---
+
+Five weeks ago we released a long awaited feature for managing database changes in Ent: **Versioned Migrations**. In
+the [announcement blog post](2022-03-14-announcing-versioned-migrations.md) we gave a brief introduction into both the
+declarative and change-based approach to keep database schemas in sync with the consuming applications, as well as their
+drawbacks and why [Atlas'](https://atlasgo.io) (Ents underlying migration engine) attempt of bringing the best of both
+worlds into one workflow is worth a try. We call it **Versioned Migration Authoring** and if you haven't read it, now is
+a good time!
+
+With versioned migration authoring, the resulting migration files are still "change-based", but have been safely planned
+by the Atlas engine. This means that you can still use your favorite migration management tool,
+like [Flyway](https://flywaydb.org/), [Liquibase](https://liquibase.org/), 
+[golang-migrate/migrate](https://github.com/golang-migrate/migrate), or 
+[pressly/goose](https://github.com/pressly/goose) when developing services with Ent.
+
+In this blog post I want to show you another new feature of the Atlas project we call the **Migration Directory 
+Integrity File**, which is now supported in Ent and how you can use it with any of the migration management tools you are
+already used to and like. 
+
+### The Problem
+
+Using versioned migration has three major downsides a developer has to be aware of in order to not break the database.
+
+1. History should not be changed
+2. The order matters
+3. SQL semantics are not checked automatically
+
+While all of the above should be detected in basic code review, it can slip the human eye, as it is error-prone.
+Therefore, an automated solution is a nice-to-have safety guard.
+
+The first issue is addressed by most management tools by saving a hash of the applied migration file to the managed
+database and comparing it with the files. If they don't match, the migration can be aborted. However, this happens in a
+very late stage in a features' development cycle, and it could save both time and resources if this can be detected
+earlier.
```

Comment:
```suggestion
The first issue (changing history) is addressed by most management tools by saving a hash of the applied migration file to the managed
database and comparing it with the files. If they don't match, the migration can be aborted. However, this happens in a
very late stage in the development cycle (during deployment), and it could save both time and resources if this can be detected
earlier.
```